src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,441 @@
+    Returns:
+        str: Name of run dir for result files to be written in.
+    """
+    timestamp = datetime.now().isoformat()
+
+    # remove any trailing slashes from user provided output_dir
+    output_dir = os.path.normpath(output_dir)
+    run_dir = f"{output_dir}/job-{timestamp}"
+
+    if not os.path.exists(output_dir) or not os.path.isdir(output_dir):
+        run_dir = f"{DEFAULTS.DK_BENCH_DATA_DIR}/job-{timestamp}"
+        logger.warning(
+            "User provided output directory path %s does not exist or is not a directory. Creating directory for results in %s",
+            output_dir,
+            DEFAULTS.DK_BENCH_DATA_DIR,
+        )
+
+    os.makedirs(run_dir, exist_ok=True)
+    logger.debug("Created directory for results at %s", run_dir)
+    return run_dir
+
+
+def print_results(result: EvaluationResult, results_files: List[str], model_name: str):
+    """
+    Prints a scoring report for DK-Bench
+    Args:
+        result (Evaluation):       ragas EvaluationResult to parse for scores
+                                   to each question.
+        results_files (List[str]): List of files with scores and responses
+        model_name (str):          Name of model that generated responses for
+                                   DK-Bench to evaluate againist reference answer.
+    Returns:
+        None
+    """
+    print("\n")
+    print("# DK-BENCH REPORT")
+    print(f"\n## MODEL: {model_name}\n")
+    total_score = 0
+    for i, score in enumerate(result.scores):
+        print(f"Question #{i+1}:     {score['domain_specific_rubrics']}/5")
+        total_score += score["domain_specific_rubrics"]
+
+    average = total_score / len(result.scores)
+    print("----------------------------")
+    print(f"Average Score:   {average:.2f}/5")
+    print(f"Total Score:     {total_score}/{len(result.scores)*5}\n")
+
+    print("Responses and scores written to:")
+    for file in results_files:
+        print(f"{file}")
+
+
+def create_excel_results_file(excel_file: str, result: EvaluationResult):
+    """
+    Writes an excel file based on the result of an evaluation.
+    The excel files has two sheets. The first is a summary sheet
+    with a table of individual question scores, average, total score and median.
+
+    The second sheet has the score, the question (user_input), reference,
+    response, model name, and evaluation run, similar to the contents of the
+    .jsonl and .csv files.
+
+    Args:
+        result (Evaluation):    ragas EvaluationResult to parse for scores
+                                for summary sheet and questions, references,
+                                and responses sheet.
+        excel_file (str):       Name of excel file to be created the summary
+                                and dataset sheets.
+    Returns:
+        None
+    """
+    summary_df = pd.DataFrame()
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
```

Comment:
@alimaredia I see that you're trying to convert an `EvaluationResult` object into a dataframe. Is there any reason why we couldn't just use the `EvaluationResult.to_pandas` method?

Reply:
Here I'm trying to create a brand new dataframe with two columns for the summary sheet. One of the columns is the question number, and then the avg,median,total score. The other column is just the scores and the values of the avg,median, score. This list is just created to populate the scores for the 2nd column.

From what I can tell this logic is tricky to review and understand so I'll take a deeper look at it to see how I could refactor or improve it.

FYI Here's an example of how the summary sheet looks in Excel:
![Screenshot from 2025-01-23 17-27-26](https://github.com/user-attachments/assets/f18be456-ca86-483d-9c17-6116629b9e68)

Reply:
@alimaredia I love what you're adding with a summary sheet. I would just remove the part where you add the score for each question. The user can just go to the other sheet if they want to look at that.

Reply:
The idea from the summary sheet is from the eval notebooks Chris wrote. From my experience it was really helpful having a question by question view so you can see how high or low the scores were that got you to your average and total. I'm leaving this for now, but what we include in the summary sheet, and all of our output files can have further thought.

Reply:
@alimaredia Can we please remove this or move it into a third sheet? The point of a summary view is to see a high level overview of where testing is at. If you already have the data on the second sheet, then there's no reason to duplicate it here. 

If the goal is to view scores without seeing all of the other information, then you can either use the built-in features of Excel to do so, or just move it into another sheet.